c++/src/sargs/SargsApplier.hh

```diff
@@ -60,10 +60,11 @@ namespace orc {
                       const std::map<uint32_t, BloomFilterIndex>& bloomFilters);
 
     /**
-     * Return a vector of bool for each row group for their selection
-     * in the last evaluation
+     * Return a vector of the next skipped row for each RowGroup. Each value is the row id
+     * in stripe. 0 means the current RowGroup is entirely skipped.
+     * Only valid after invoking pickRowGroups().
      */
-    const std::vector<bool>& getRowGroups() const { return mRowGroups; }
```

Comment:
There is no way to keep the original method? This looks like a breaking change if we remove this public method.

Reply:
IIUC, _**SargsApplier.hh**_ is not a public header file.

Reply:
I think all the public header files that our users can use are https://github.com/apache/orc/tree/main/c%2B%2B/include/orc
So this is not a breaking change.

BTW, the current `vector<uint64_t>` result is a superset of the original `vector<bool>` in the way that 0 values mean not selected and non-zero values mean selected. If your downstream projects port the ORC codes directly and depend on this method, you can simply change the usages in this way.

If this is still not good enough, we can choose the simpler fix proposed by @wgtmac.

Reply:
I'm fine with the AS-IS PR if you folks are sure. :)